tools/export_to_path/export_to_path.py

```diff
@@ -1,53 +1,44 @@
 #!/usr/bin/env python
 
 from __future__ import print_function
+
 import optparse
 import os
 import os.path
-import re
 import shutil
 import sys
 
 '''
 Inspired by @nsoranzo's https://github.com/TGAC/earlham-galaxytools/tree/master/tools/export_to_cluster/
 '''
-
+SALLY_MOUNT_PREFIX = "/mnt/sally/000020-Shares/rcx-da"
 exit_code = 0
-dir_prefix = "/mnt/sally"
-dir_suffix = 'mzML_profile'
 parser = optparse.OptionParser()
-parser.add_option('-d', '--export_dir', help='Directory where to export the datasets')
+parser.add_option('-p', '--remote_path', help='Remote path to mzML')
+#  remote_path = "/mnt/sally/000020-Shares/rcx-da/H2020_HBM4EU/2020/WP16_Specimen/HBM4EU_ESI_positive_WP_urine_MS1/HBM4EU_Fieldwork_1_Batch_1-20201001-CMV/mzML_profile/Tribrid_201001_051-350697_POS_MU.mzML"
 (options, args) = parser.parse_args()
-if not options.export_dir:
-    parser.error('Export directory cannot be empty')
-if not args or (len(args) % 3) != 0:
-    raise Exception("Incorrect number of arguments specified.")
-full_export_dir = os.path.join(dir_prefix, options.export_dir.lstrip(os.sep), dir_suffix)
-real_export_dir = os.path.realpath(full_export_dir)
-if not real_export_dir.startswith(dir_prefix):
-    raise Exception("'%s' must be a subdirectory of '%s'" % (real_export_dir, dir_prefix))
-if not os.path.exists(real_export_dir):
-    os.makedirs(real_export_dir)
-dataset_paths = args[::3]
-dataset_names = args[1::3]
-dataset_exts = args[2::3]
-
-for dp, dn, de in zip(dataset_paths, dataset_names, dataset_exts):
-    """
-    Copied from get_valid_filename from django
-    https://github.com/django/django/blob/master/django/utils/text.py
-    """
-    dn_de = "%s.%s" % (dn, de)
-    dn_de_safe = re.sub(r'(?u)[^-\w.]', '', dn_de.strip().replace(' ', '_'))
-    dest = os.path.join(real_export_dir, dn_de_safe)
-    if os.path.isfile(dest):
-        raise Exception("Error copying dataset '%s' to '%s'. Cannot overwrite existing dataset" % (dn, dest))
+if not options.remote_path:
+    parser.error('Remote path cannot be empty.')
+export_dir, export_file = os.path.split(options.remote_path)
+if not os.path.exists(export_dir):
+    os.makedirs(export_dir)
+#  Make sure program will never write outside of the mounted target directory.
+if not os.path.commonpath([os.path.realpath(options.remote_path), SALLY_MOUNT_PREFIX]) == SALLY_MOUNT_PREFIX:
+    raise Exception(f"Invalid export path supplied: {options.remote_path}")
+remote_paths = [options.remote_path]
+if len(args) > 1:
+    remote_paths.append(f"{options.remote_path[:-5]}.json")
+if len(args) == 3:
+    remote_paths.append(f"{options.remote_path[:-5]}.txt")
```

Comment:
Could you explain what these checks exactly are in a comment (maybe in the code)? To me they seem a bit magic from just having a quick look - why add txt if `len(args) == 3` and json if `len(args) > 1` ?

Reply:
I agree it is not very robust, the idea was to have positional arguments that would be used with json metadata option as default but would also work when eventually we have thermo that outputs both.

I also have a branch that narrows this down to "copy one dataset to a given path". Since we control all the paths outside of Galaxy it might be the easier way forward.